Board Pine H6 add RTL8723

Parche original de la distribución de Armbian, por martinayotte: https://github.com/armbian/build/commit/1e8aa126227d84bc107215853002637d24139174#diff-625d13b86b549a8b48b52df57733b848

AlexTMjugador: editado nombre del nodo del NIC de WLAN por ser incorrecto: el chip es un RTL8723BS, no un RTL8723CS. Está grabado el modelo en la propia carcasa del IC.
También he eliminado algunas declaraciones redundantes en el nodo mmc1, que ya se heredan de sun50i-h6.dtsi.

Signed-off-by: AlexTMjugador <AlexTMjugador@users.noreply.github.com>
---
diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts-old b/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts
index 52a1574..326195b 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts-old
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts
@@ -75,6 +75,14 @@
 		enable-active-high;
 	};
 
+	wifi_pwrseq: wifi_pwrseq {
+		compatible = "mmc-pwrseq-simple";
+		clocks = <&rtc 1>;
+		clock-names = "ext_clock";
+		reset-gpios = <&r_pio 1 3 GPIO_ACTIVE_LOW>; /* PM3 */
+		post-power-on-delay-ms = <200>;
+	};
+
 	leds {
 		compatible = "gpio-leds";
 
@@ -160,6 +168,22 @@
 	status = "okay";
 };
 
+&mmc1 {
+	vmmc-supply = <&reg_cldo2>;
+	vqmmc-supply = <&reg_bldo2>;
+	mmc-pwrseq = <&wifi_pwrseq>;
+	bus-width = <4>;
+	non-removable;
+	status = "okay";
+
+	rtl8723bs: sdio_wifi@1 {
+		reg = <1>;
+		interrupt-parent = <&r_pio>;
+		interrupts = <1 0 IRQ_TYPE_LEVEL_LOW>; /* PM0 */
+		interrupt-names = "host-wake";
+	};
+};
+
 &mmc2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&mmc2_pins>;
@@ -263,12 +263,24 @@
 			};
 
 			reg_cldo2: cldo2 {
+				/*
+				 * This regulator is connected with CLDO3.
+				 * Before the kernel can support synchronized
+				 * enable of coupled regulators, keep them
+				 * both always on as a ugly hack.
+				 */
+				regulator-always-on;
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-name = "vcc-wifi-1";
 			};
 
 			reg_cldo3: cldo3 {
+				/*
+				 * This regulator is connected with CLDO2.
+				 * See the comments for CLDO2.
+				 */
+				regulator-always-on;
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-name = "vcc-wifi-2";
